### PR TITLE
 Enhancement of xml handling

### PR DESCRIPTION
**Pull request by Maxmawt**

### CODE REVIEW

File: inginious-problems-blockly/__init__.py [hunk 1]
```diff
@@ -137,17 +137,19 @@ def adapt_input_for_backend(self, input_data):
...
+        blockly_dico["taskid"] = task.get_id()
+        blockly_dico["filenames"] = []
         for filename in self._files + self._blocks_files:
-            filenames.append(str(filename))
-        toolbox = self._toolbox
-        return str(DisplayableBlocklyProblem.get_renderer(template_helper).box_blockly(courseid, taskid, self.get_id(),
-                                                                                self.get_name(), toolbox,
-                                                                                filenames, self._workspace,
-                                                                                json.dumps(self._options)))
+            blockly_dico["filenames"].append(str(filename))
```
Comment (anthonygego) [inline inginious-problems-blockly/__init__.py]: Can't you use list comprehension here ?

**Comment by anthonygego** [comment]
I suggest that you learn how to rebase your code to avoid doing squash-merges on github all the time to keep the commit history clean. I think that one commit is enough for such a PR.

**Comment by Maxmawt** [comment]
Yes, that should be a great idea

**Comment by anthonygego** [comment]
Any update ?

**Comment by Maxmawt** [comment]
Sorry, I forgot ;)

**Comment by Maxmawt** [comment]
better? or you would like more rebase?

**Comment by anthonygego** [comment]
>I think that one commit is enough for such a PR.

Squash everything. Your branch should also be rebased from master to avoid several blank commits from Bastien.

In summary your PR should only display one commit :-)

**Comment by Maxmawt** [comment]
Ok for you?

**Pull request merged**